resources/d.rb

```diff
@@ -17,72 +17,87 @@
-property :minute, [Integer, String], default: '*', callbacks: {
-  'should be a valid minute spec' => lambda do |spec|
-    ::Cron::Helpers.validate_numeric(spec, 0, 59)
-  end,
-}
-property :hour, [Integer, String], default: '*', callbacks: {
-  'should be a valid hour spec' => ->(spec) { ::Cron::Helpers.validate_numeric(spec, 0, 23) },
-}
-property :day, [Integer, String], default: '*', callbacks: {
-  'should be a valid day spec' => ->(spec) { ::Cron::Helpers.validate_numeric(spec, 1, 31) },
-}
-property :month, [Integer, String], default: '*', callbacks: {
-  'should be a valid month spec' => ->(spec) { ::Cron::Helpers.validate_month(spec) },
-}
-property :weekday, [Integer, String], default: '*', callbacks: {
-  'should be a valid weekday spec' => ->(spec) { ::Cron::Helpers.validate_dow(spec) },
-}
-
-property :command, String, required: true
-property :user, String, default: 'root'
-property :mailto, [String, NilClass]
-property :path, [String, NilClass]
-property :home, [String, NilClass]
-property :shell, [String, NilClass]
-property :comment, [String, NilClass]
-property :environment, Hash, default: {}
-property :mode, [String, Integer], default: '0600'
-property :random_delay, [Integer, NilClass]
-
-def after_created
-  raise 'The cron_d resource requires Linux as it needs support for the cron.d directory functionality.' unless node['os'] == 'linux'
-end
+unified_mode true
 
-action :create do
-  create_template(:create)
-end
+property :cron_name, String,
+          name_property: true
 
-action :create_if_missing do
-  create_template(:create_if_missing)
-end
+property :cookbook, String,
+          default: 'cron'
 
-action :delete do
-  # cleanup the legacy named job if it exists
-  file 'legacy named cron.d file' do
-    path "/etc/cron.d/#{new_resource.cron_name}"
-    action :delete
-  end
+property :predefined_value, String,
+          equal_to: %w( @reboot @yearly @annually @monthly @weekly @daily @midnight @hourly )
 
-  file "/etc/cron.d/#{sanitized_name}" do
-    action :delete
-  end
-end
+property :minute, [Integer, String],
+          default: '*',
+          callbacks: {
+                        'should be a valid minute spec' => ->(spec) { ::Cron::Cookbook::CronHelpers.validate_numeric(spec, 0, 59) },
+                      }
+
+property :hour, [Integer, String],
+          default: '*',
+          callbacks: {
+                       'should be a valid hour spec' => ->(spec) { ::Cron::Cookbook::CronHelpers.validate_numeric(spec, 0, 23) },
+                     }
+
+property :day, [Integer, String],
+          default: '*',
+          callbacks: {
+                       'should be a valid day spec' => ->(spec) { ::Cron::Cookbook::CronHelpers.validate_numeric(spec, 1, 31) },
+                     }
+
+property :month, [Integer, String],
+          default: '*',
+          callbacks: {
+                       'should be a valid month spec' => ->(spec) { ::Cron::Cookbook::CronHelpers.validate_month(spec) },
```

Comment:
We can probably just use: `equal_to: [1,2,3,4,jan,feb]` instead of the helper maybe?

Happy either way.

Reply:
I think it's best kept as the helper as we'd end up with a huuuuuge array by the time you've covered the jan-dec 1-12(string) 1-12(int) and '*'.

Reply:
I've just carried them over from before as the best thing for this cookbook is really for it all to go away as these are core chef resources in 14.something. I've only updated this because it's used by a few of the older chef-cookbook cookbooks.